cheatsheets/Forgot_Password_Cheat_Sheet.md

```diff
@@ -1,60 +1,137 @@
 
-The user should always be sent a token rather than a new password for their account. There are numerous problems with directly sending the user a password, especially via email. These include:
+### User Resets Password
 
-- The transport mechanism used to send the password (email, SMS, etc) may not be encrypted.
-- The password may be stored in the users' mailbox or device for an indefinite period.
-- The password may be accessible to IT staff or other users who have access to the mailbox or device.
-- The password may be logged by third parties such as mail security scanning platforms.
+Once the user is validated through the provided token or code, they should reset their password to a new secure one. In order to secure this last step, the measures should be taken:
 
-Of course, by all means, once a user's password has been reset, the randomly-generated token should no longer be valid.
+- Validate that a secure password policy is in place.
+- The user should confirm the password they set by writing it twice.
+- Update and store the password following [secure practices](Password_Storage_Cheat_Sheet.md).
+- Send the user an email informing them that their password has been reset.
+- Ask the user to re-login. Don't auto-login users on password reset!
+- Ask the user if they want to invalidate all of the sessions, or invalidate the sessions by default without prompting the user.
 
-## Step 4) Allow user to change password in the existing session
+## Methods
 
-Step 4 requires input of the code sent in step 3 in the existing session where the challenge questions were answered in step 2, and allows the user to reset his password. Display a simple HTML form with one input field for the code, one for the new password, and one to confirm the new password. Verify the correct code is provided and be sure to enforce all password complexity requirements that exist in other areas of the application. 
+In order to allow a user to request a password reset, you will need to have some way to identify the user, or a means to reach out to them through a side-channel.
 
-As before, avoid sending the username as a parameter when the form is submitted. Finally, it's critical to have a check to prevent a user from accessing this last step without first completing steps 1 and 2 correctly. Otherwise, a [forced browsing](https://owasp.org/www-community/attacks/Forced_browsing) attack may be possible. Ensure the user changes their password and does not simply surf to another page in the application. 
+This can be done through any of the following methods:
 
-The reset must be performed before any other operations can be performed by the user.
+- One Time Password (OTP).
+- URL tokens.
+- Security Questions.
+- Offline backup codes.
 
-## Step 5) Logging
+> These methods can be used together and many times it is recommended to do so. No matter what you must ensure that a user always has a way to recover their account.
```

Comment:
Are you quoting this from somewhere? If so, reference it. If not, remove the `> ` quote marker